rfcs/text/testing-refinement.md

```diff
@@ -0,0 +1,143 @@
+- Start Date: 2020-03-31
+
+
+# Table of contents
+
+- [Table of contents](#table-of-contents)
+- [Glossary](#glossary)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Visual Comparison testing](#visual-comparison-testing)
+  - [Reformatting of component stories for testing](#reformatting-of-component-stories-for-testing)
+- [Detailed design](#detailed-design)
+  - [Test stories](#test-stories)
+  - [Test feature files](#test-feature-files)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Adoption strategy](#adoption-strategy)
+- [How we teach this](#how-we-teach-this)
+- [Unresolved questions](#unresolved-questions)
+
+
+# Glossary
+
+| Term           | Description                                                                    |
+|----------------|--------------------------------------------------------------------------------|
+| Applitools     | An automated visual UI testing tool                                            |
+| canvas         | An area in the preview panel in Storybook                                      |
+| CSS            | Cascading Style Sheets, to describe how HTML elements are to be displayed      |
+| Cypress        | An open source frontend testing framework                                      |
+| Design System  | Guidelines around use, styles, behaviours etc for developing Carbon components |
+| knobs          | A means of editing component props dynamically in the Storybook UI             |
+| props          | Properties of the component which affect how it appears/behaves when rendered  |
+| Storybook      | An open source frontend tool for developing and organising UI components       |
+
+# Summary
+
+The Carbon team would like to use a more rapid and accurate method of regression testing Carbon components to reduce development to release time and enable the regression suite to be run at every commit.
+
+# Motivation
+
+The regression suite uses the [Cypress.io](https://www.cypress.io) framework to test all knob configurations, events and accessibility of all stories of all components in Storybook. In its current form it takes around 2.5 hours to run. When the build itself only takes around 7 minutes, it is disproportionate to have a regression suite for those same components to take that much longer to run by such a huge factor. We require a method by which the visual UI can be tested rapidly, accurately and at every commit to detect regressions earlier.
```

Comment:
according to [this change](https://github.com/Sage/carbon/pull/2880) regression time will be under 30 min

Reply:
The problem is that these two pieces of work have been done in parallel, so any changes I make in this RFC are out of date when changes are done in the other stream. I'm going to change this section altogether as the motivation for integrating Applitools now is no longer to reduce the time taken, only to provide the visual UI testing.